vm-setup/roles/common/tasks/write_ironic_nodes_tasks.yml

```diff
@@ -0,0 +1,10 @@
+---
+# Generate the ironic node inventory files.  Note that this
+# task *must* occur after the above vm tasks, because if
+# `vm_nodes` is defined the template depends on the
+# `node_mac_map` variable.
```

Comment:
nit: This comment makes sense where the task is used but not here where it is defined.